src/Libraries/CoreNodes/GeometryColor.cs

```diff
@@ -60,25 +60,65 @@ public void Tessellate(IRenderPackage package, double tol = -1, int maxGridLines
             var surf = geometry as Surface;
             if (surf != null)
             {
+                var start = package.LineStripVertexColors.Count;
                 surf.PerimeterCurves().ForEach(
                         e =>
                             e.Tessellate(
                                 package,
                                 tol,
                                 maxGridLines));
+                var end = package.LineStripVertexColors.Count - 1;
+                ReColorVerticesFromTo(start, end, package);
             }
 
             var solid = geometry as Solid;
             if (solid != null)
             {
+                var start = package.LineStripVertexColors.Count;
                 solid.Edges.ForEach(
                         e =>
                             e.CurveGeometry.Tessellate(
                                 package,
                                 tol,
                                 maxGridLines));
+                var end = package.LineStripVertexColors.Count - 1;
+                ReColorVerticesFromTo(start, end, package);
             }
-    }
+
+            for (var i = 0; i < package.TriangleVertices.Count; i += 3)
+            {
+                NudgeVertexAlongVector(package.TriangleVertices, package.TriangleNormals, i, 0.001);
+            }
+
+        }
+
+        private void NudgeVertexAlongVector(IList<double> vertices, IList<double> normals, int i, double amount)
```

Comment:
Having this a geometry shader is a really good idea. The only difficulty is that then I have to shuffle different bits of geometry into different collections to be drawn with different shaders. Right now everything is drawn with one shader. There is definitely room for future improvement.